### DISCUSSION TITLE
DI and Simple Code

### DISCUSSION THREAD

**Discussion by ShervanN** [ProfessionalCSharp/ProfessionalCSharp2021#185] [Q&A] [answered]
Hi,

I wrote the following code to learn Dependency Injection.
Is it true? Could you please check this code?

Thanks

```
namespace ForTest
{
    interface IEmail
    {
        public void Send();
    }
    class Email
    {
        private IEmail _iEmail;
        public Email(IEmail iEmail)
        {
            _iEmail = iEmail;
        }
        public void SentEmails()
        {
            _iEmail.Send();
        }
    }
    class EmailService : IEmail
    {
        public void Send()
        {
            Console.WriteLine("Sent!");
        }
    }
    internal class Program
    {
        static void Main(string[] args)
        {
            Email myEmail = new Email(new EmailService());
            myEmail.SentEmails();

            Console.ReadKey();
        }

    }
}
```

**Comment by christiannagel** [comment, marked as answer]
What do you mean by "is it true"? 
Regarding dependency injection - yes, you inject the `EmailService` wich implements `IEmail` into the `Email` class. You can create different providers implementing `IEmail` (e.g. a test provider, a provider using SMPT, a provider using Office 365...), and use all these with your `Email` class. 

I just would rename the `IEmail` interface to `IEmailSender`. With `IEmail` I would think more about the email content such as addresses, header, content...
